### PR TITLE
Fix bug in LiveTradingRealTimeHandler

### LINKED ISSUE #7662: Live Trading Market Hours Refresh
#### Expected Behavior

- For long live deployments (years) Holidays/EarlyCloses/LateOpens might of been added or removed to the MHDB, and ideally Lean will pick up those updates
#### Actual Behavior

- `LiveTradingRealTimeHandler` is running a daily `RefreshMarketHoursToday` but Holidays/EarlyCloses/LateOpens aren't getting updated
#### Potential Solution

Expand `LiveTradingRealTimeHandler.RefreshMarketHoursToday` to refresh Holidays/EarlyCloses/LateOpens too
#### Reproducing the Problem

N/A
#### System Information

N/A
#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by Marinovsky**
#### Description

So far, `LiveTradingRealTimeHandler` didn't take into consideration holidays nor early closes/late opens when refreshing the market hours for the Securities. In the same way, if certain update to the MHDB(MarketHoursDatabase) was done, it didn't upload its instance of the MHDB. This PR solves those bugs.
#### Related Issue

Close #7662 
#### Motivation and Context

With this change, users will get accurate market hours for their securities when using `LiveTradingRealTimeHandler`, even after the MHDB is updated. 
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

There were created unit tests asserting the market hours assigned to the algorithm's securities took into consideration holidays, early closes and late opens. Additionally, it was asserted that the MHDB reset every time the method `RefreshMarketHoursToday()` was called.
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by jhonabreul — CHANGES_REQUESTED** [review]
Thanks! 🚀  Leaving some minor commets here. Shared some others through Slack

File: Tests/Engine/RealTime/LiveTradingRealTimeHandlerTests.cs [hunk 1]
```diff
@@ -78,5 +163,154 @@ public bool TryRequestAdditionalTime(int minutes)
...
+                    }
+                    Assert.AreEqual(expectedSegment.End, security.Exchange.Hours.GetNextMarketClose(time.Date, false).TimeOfDay);
+                    Assert.AreEqual(expectedSegment.Start, security.Exchange.Hours.GetNextMarketOpen(time.Date, false).TimeOfDay);
+                }
+
+                Exit();
+            }
+        }
+
+        public class TestLiveTradingRealTimeHandlerReset : LiveTradingRealTimeHandler
```
Comment (jhonabreul) [inline Tests/Engine/RealTime/LiveTradingRealTimeHandlerTests.cs:215]: Make this test classes private

File: Tests/Engine/RealTime/LiveTradingRealTimeHandlerTests.cs [hunk 2]
```diff
@@ -63,6 +74,80 @@ public void ThreadSafety()
...
+            algorithm.SetEndDate(2023, 5, 30);
+            var security = algorithm.AddEquity("SPY");
+            security.Exchange = new SecurityExchange(SecurityExchangeHours.AlwaysOpen(TimeZones.NewYork));
+            var symbol = security.Symbol;
+            algorithm.SetFinishedWarmingUp();
+
+            var handleOptionNotification = typeof(BrokerageTransactionHandler).GetMethod("HandleOptionNotification", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            var transactionHandler = new TestBrokerageTransactionHandler();
+            var broker = new BacktestingBrokerage(algorithm);
```
Comment (jhonabreul) [inline Tests/Engine/RealTime/LiveTradingRealTimeHandlerTests.cs:116]: This needs disposing

**Review by jhonabreul — CHANGES_REQUESTED** [review]

File: Tests/Engine/RealTime/LiveTradingRealTimeHandlerTests.cs [hunk 3]
```diff
@@ -78,5 +163,154 @@ public bool TryRequestAdditionalTime(int minutes)
...
+            public void SetMarketHoursDatabase(MarketHoursDatabase marketHoursDatabase)
+            {
+                MarketHoursDatabase = marketHoursDatabase;
+            }
+
+            public void TestRefreshMarketHoursToday(Security security, DateTime time, MarketHoursSegment expectedSegment)
+            {
+                OnSecurityUpdated.Reset();
+                RefreshMarketHoursToday(time);
+                OnSecurityUpdated.WaitOne();
```
[the remark this answers is not in the document]
Comment (Marinovsky) [reply Tests/Engine/RealTime/LiveTradingRealTimeHandlerTests.cs:179]: Yes, it does 😅 since it calls `GetMarketHours()`

**Review by jhonabreul — APPROVED** [review]
Thanks 👍 Looks good. Minor comment shared through Slack

**Pull request merged**